### DISCUSSION TITLE
Add traceId to the log

### DISCUSSION THREAD

**Discussion by mitht** [open-telemetry/opentelemetry-java-instrumentation#10663] [Q&A]
Automatic Instrumentation is used
How to splice the traceId into the log content without modifying the system and causing zero intrusion into the business code

**Comment by jaydeluca** [comment]
I think it will depend on what you are using for logging, but have you looked into using the MDC?
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/logger-mdc-instrumentation.md

**Reply by mitht** [reply]
Thanks for your answer
Pattern must be added to the logback.xml of the business system.

<property name="pattern" value="%d [%thread] %-5p [%c] [%F:%L] [trace=%X{trace_id:-},span=%X{span_id:-}] - %msg%n"/>


Is there any way without changing the business system?

**Reply by laurit** [reply]
You could export your logs via otlp, then the exported information will contain the trace and span id. Alternatively you can write your own extension https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/extension that instruments your logging framework and adds whatever you need to the output.
